November 2024 Newsletter (#838)
RSE-Sheffield/RSE-Sheffield.github.io: _newsletters/2024-11-01-newsletter.md

---
layout: post
title: November 2024 Newsletter
editor: Matthew Leach
slug: 2024-11-01-newsletter
date: 2024-11-01T12:00:00Z+0100
tags:
category:
link:
description:
type: text
---

## The University of Sheffield Research Software Engineering Community Newsletter November 2024

Welcome to the November 2024 newsletter for the research software community at The University of Sheffield, featuring news, opportunities, events and training for you.

### RSE FAIR<sup>2</sup> 4RS Training Programme Launch!

> This training curriculum offers a modular programme to support researchers in applying FAIR (Findable, Accessible, Interoperable, Reusable) principles and open research practices to their research software. The programme is aimed at researchers, including PhDs and postgraduate research students, who create code (whether a few scripts or something more substantial) as part of their research and who want to make their research more open by applying the FAIR principles to their software or simply want to become more confident in the research code they are writing.

Reviewer: It would be worth linking to the original FAIR4RS principles I think 
```suggestion
This training curriculum offers a modular programme to support researchers in applying FAIR (Findable, Accessible, Interoperable, Reusable) principles and open research practices to their [research software](https://zenodo.org/records/6623556). The programme is aimed at researchers, including PhDs and postgraduate research students, who create code (whether a few scripts or something more substantial) as part of their research and who want to make their research more open by applying the FAIR principles to their software or simply want to become more confident in the research code they are writing.
```

> This training curriculum offers a modular programme to support researchers in applying FAIR (Findable, Accessible, Interoperable, Reusable) principles and open research practices to their research software. The programme is aimed at researchers, including PhDs and postgraduate research students, who create code (whether a few scripts or something more substantial) as part of their research and who want to make their research more open by applying the FAIR principles to their software or simply want to become more confident in the research code they are writing.

Reviewer: Should perhaps also clarify that the FAIR^2 training is for researchers at the University of Sheffield...

```suggestion
This training curriculum offers a modular programme to support researchers in applying FAIR (Findable, Accessible, Interoperable, Reusable) principles and open research practices to their research software. The programme is aimed at researchers, including PhDs and postgraduate research students at The University of Sheffield, who create code (whether a few scripts or something more substantial) as part of their research and who want to make their research more open by applying the FAIR principles to their software or simply want to become more confident in the research code they are writing.
```

> The programme will be running over the next few months, check out for [more information](https://rse.shef.ac.uk/training/fair4rs/) or to [sign up](https://rse.shef.ac.uk/training/fair4rs/) for sessions!

Reviewer: "_check out for more information_" reads funny in my head and both links point to the same URL on the RSE website, perhaps we should just direct them to that as courses that have been schedule have registration links
```suggestion
The programme will be running over the next few months. For more information  check out for [the RSE page](https://rse.shef.ac.uk/training/fair4rs/) where you will find links to register for modules that are already scheduled. You can also find the modules on [myDevelopment](https://mydevelopment.csod.com/ui/lms-learning-details/app/curriculum/06573f55-12e4-44f1-be4c-fac5ea9d6851) (requires institutional login).
```

> In the first session which took place in October, our Head of Research Software Engineering, Romain Thomas, delivered a talk titled "Better software for Better research: Introduction to the FAIR2 for Research Software training programme". In this introductory talk we present what the FAIR principles are and why they are crucial to make your research more open. We give some initial hints on how to apply them to software and present the FAIR training programme that our teams have designed for the UoS research community.

Reviewer: Tweaking to past tense.

```suggestion
In the first session which took place in October, our Head of Research Software Engineering, Romain Thomas, delivered a talk titled "Better software for Better research: Introduction to the FAIR2 for Research Software training programme". In this introductory talk he presented what the FAIR principles are and why they are crucial to make your research more open. We gave some initial hints on how to apply them to software and present the FAIR training programme that our teams have designed for the UoS research community.
```

The [recording](https://orda.shef.ac.uk/articles/media/Better_software_for_Better_research_Introduction_to_the_FAIR2_for_Research_Software_training_programme/27283239?file=49937889) of the talk and the [slides](https://fair2-for-research-software.github.io/Better_software_for_better_research/#/title-slide) are both available.

### News

- [Open consultation Invest 2035: the UK's modern industrial strategy](https://www.gov.uk/government/consultations/invest-2035-the-uks-modern-industrial-strategy/invest-2035-the-uks-modern-industrial-strategy): The new modern industrial strategy – Invest 2035 – is the UK government’s credible, 10-year plan to deliver the certainty and stability businesses need to invest in the high growth sectors that will drive our growth mission. This green paper sets out the government’s approach and asks for your views and evidence to help develop a successful, modern industrial strategy to be launched in spring 2025.

> - [Bede: 3 additional Grace-Hopper nodes available (6 total)](https://bede-documentation.readthedocs.io/en/latest/usage/index.html#grace-hopper-pilot): The N8 CIR Bede Tier 2 HPC facility now includes a pilot of 6 Grace-Hopper nodes, each containing an NVIDIA GH200 Grace Hopper Superchip. Bede is available for use by researchers from N8 Research Partnership, including the University of Sheffield. Access to Bede is managed per project, with details of the [application procedure](https://n8cir.org.uk/bede/accessing-bede/) provided on the N8 CIR website. Please contact `tier-2-hpc-support-group@sheffield.ac.uk` if you have any questions regarding the application process, the Grace-Hopper pilot, or Bede in general.

Reviewer: `mailto:` link for email address...

```suggestion
- [Bede: 3 additional Grace-Hopper nodes available (6 total)](https://bede-documentation.readthedocs.io/en/latest/usage/index.html#grace-hopper-pilot): The N8 CIR Bede Tier 2 HPC facility now includes a pilot of 6 Grace-Hopper nodes, each containing an NVIDIA GH200 Grace Hopper Superchip. Bede is available for use by researchers from N8 Research Partnership, including the University of Sheffield. Access to Bede is managed per project, with details of the [application procedure](https://n8cir.org.uk/bede/accessing-bede/) provided on the N8 CIR website. Please contact [tier-2-hpc-support-group@sheffield.ac.uk](mailto:tier-2-hpc-support-group@sheffield.ac.uk) if you have any questions regarding the application process, the Grace-Hopper pilot, or Bede in general.
```

- Until Nov 10th - [MATLAB Shorts Mini Hack](https://uk.mathworks.com/matlabcentral/contests/2024-matlab-mini-hack.html): Use upto 2,000 characters of MATLAB code to generate a MATLAB movie and win prizes.
- Thursday's 1pm - [Cambridge RSE Seminar series](https://rse.group.cam.ac.uk/seminars) (currently scheduled till Dec 5th)
- April 29-30th 2025 - [UK Technology Specialists Network Conference](https://itss.org.uk/uk-technology-specialists-network-conference-2025/) 

### Articles, Blogs, Papers & Podcasts

#### Articles & Blogs
- [Pytest Matplotlib](https://blog.nshephard.dev/posts/pytest-mpl/): The RSE team's Neil Shephard writes about his experience of using pytest-mpl to provide automated testing of functions which generate images using matplotlib or similar libraries.
- [Winamp deletes entire GitHub source code repo after a rocky few weeks](https://arstechnica.com/gadgets/2024/10/winamp-really-whips-open-source-coders-into-frenzy-with-its-source-release/): A case study on the importance of licencing!

#### Papers

> - *Pre-print* [Software Frugality in an Accelerating World: the Case of Continuous Integration](https://arxiv.org/abs/2410.15816)

Reviewer: _Italics_ or **Bold** ? Renders as the former but bold probably stands out clearer.

- *Pre-print* [Extending and Applying Automated HERMES Software Publication Workflows](https://arxiv.org/abs/2410.17614)

#### Podcasts
- [HPC software installations made EESSI - Alan O'Cais and Kenneth Hoste](https://codeforthought.buzzsprout.com/1326658/episodes/15867365-en-hpc-software-installations-made-eessi-alan-o-cais-and-kenneth-hoste): Meet Kenneth Hoste and Alan O'Cais who will tell us about EESSI,The European Environment for Scientific Software Installation for high performance computing (HPC). 

#### Books
- [Research Software Engineering - A Guide to the Open Source Ecosystem](https://rse-book.github.io/): This book aims at two things: First, to give a big picture overview and starting point to reach what the open source software community calls a “software carpentry” level. Second, to give an understanding of the opportunities of automation and reproducibility, as well as the effort to maintain the required environment. 

### Training

#### At Sheffield

> **FAIR<sup>2</sup> 4RS Workshops**

Reviewer: As per below I can see why this might be a standing section in all newsletters, but wonder if in this newsletter its worth highlighting these as already being scheduled in the introduction.

Author: I think it's helpful to have both the introductory promotion of the programme overall and the individual upcoming events listed here to maximise coverage, but I think your idea of highlighting that they are part of the programme discussed in the introduction is a good one, I will add that in

- 18th & 19th November - [Version Control](https://rse.shef.ac.uk/training/fair4rs/#version-control): The version control module has two distinct training sessions: one for beginners and one for more advanced users. The first introduces key version control concepts such as forks, pull requests and branches using Git, GitHub and GitKraken, then gives you the chance to get some hands-on experience with using version control in a research setting. The second session aims to help you develop a deeper understanding of how Git works to facilitate collaboration by improving your understanding of working with branches and how to make your commits tidier, as well as understanding how to effectively use pull requests and Git history.
- 9th & 10th December - [Software Design](https://rse.shef.ac.uk/training/fair4rs/#software-design): The way you write your code will have a massive impact on how easy it is to maintain. During this course we will learn how to create maintainable, readable and reusable code.

> The [FAIR<sup>2</sup> for Research Software Training Programme](https://rse.shef.ac.uk/training/fair4rs/) has been developed

Reviewer: Feels like duplication of the introductory section but I can see why this might be a standing section common to all newsletters.

> by the RSE, Data Analytics Service and Library teams and will be launched in October. The full syllabus is now live on [MyDevelopment](https://mydevelopment.csod.com/ui/lms-learning-details/app/curriculum/06573f55-12e4-44f1-be4c-fac5ea9d6851) (requires login).

Reviewer 2: October has passed.

Author: Good spot, resolved now

- 26th - 27th November [Archer 2 - Message Passing programming with MPI](https://www.archer2.ac.uk/training/courses/241126-mpi/): The world’s largest supercomputers are used almost exclusively to run applications which are parallelised using Message Passing. The course covers all the basic knowledge required to write parallel programs using this programming model, and is directly applicable to almost every parallel computer architecture.
- 13th November [Reproducible analysis: Python for social scientists](https://www.software.ac.uk/news/registration-now-open-second-round-research-software-practices-social-sciences-workshops): introduce Python as a coding language and useful analytical tool in the social sciences studies. During the workshop, the instructor will practically demonstrate (i.e. through live coding) the basics of this coding language, including key points of the standard Python workflow like exploring datasets, performing basic summary statistics and visualisations, and exporting project results. 

### Funding Opportunities

> - [FAIR-IMPACT's 3rd call](https://fair-impact.eu/3rd-open-call-route-2-support-opens-30-sept), deadline 4th Dec, hopes to fund applicants in four areas

Reviewer: Pedantic but could list these in chronological order based on deadlines. Perhaps highlight the deadlines too.
```suggestion
- [FAIR-IMPACT's 3rd call](https://fair-impact.eu/3rd-open-call-route-2-support-opens-30-sept), **deadline 4th Dec**, hopes to fund applicants in four areas
```

#### Jobs

Check for advertised RSE and RSE-adjacent roles at the [RSE society's vacancies
board](https://society-rse.org/careers/vacancies/).

### Community

#### Digital Research Practice Support Community

The DRPS community is a group for people that support researchers in carrying out research in the digital age. Meetings
are held monthly, with discussions around events, training and opportunities related to the field.

You can join the google group
[here](https://groups.google.com/u/1/a/sheffield.ac.uk/g/digital-research-practice-support-community-group/about) to
stay informed.

The next meeting is scheduled for 2pm on Wednesday 6th November 2024.

#### LunchBytes

[LunchBytes](https://rse.shef.ac.uk/community/lunch-bytes/) are short talks from the research community on research
software, data, and infrastructure.

Due to capacity within the RSE team, there was very limited LunchBytes content over the past academic year.
This should be changing soon, a new coordinator for LunchBytes has been found (Research IT's Farhad Allian), with a plan to reboot these lunchtime seminars in the spring semester.
 

> More information will be coming out over the coming months, so keep your eyes peeled if you'd like to learn more about research software and associated practices, or share what you know!

Reviewer: ```suggestion
More information will be coming out over the coming months, so sign up to the [RSE mailing list](https://groups.google.com/a/sheffield.ac.uk/g/RSE-group) if you'd like to learn more about research software and associated practices, or get in contact with [Farhad](mailto:farhad.allian@sheffield.ac.uk) if you would like to share what you know at an upcoming session!
```

### RSE FAIR<sup>2</sup> 4RS Training Programme Launch!
This training curriculum offers a modular programme to support researchers in applying FAIR (Findable, Accessible, Interoperable, Reusable) principles and open research practices to their [research software](https://zenodo.org/records/6623556). The programme is aimed at researchers, including PhDs and postgraduate research students at the University of Sheffield, who create code (whether a few scripts or something more substantial) as part of their research and who want to make their research more open by applying the FAIR principles to their software or simply want to become more confident in the research code they are writing.

The programme will be running over the next few months. For more information  check out for [the RSE page](https://rse.shef.ac.uk/training/fair4rs/) where you will find links to register for modules that are already scheduled. You can also find the modules on [myDevelopment](https://mydevelopment.csod.com/ui/lms-learning-details/app/curriculum/06573f55-12e4-44f1-be4c-fac5ea9d6851) (requires institutional login).

> In the first session which took place in October, our Head of Research Software Engineering, Romain Thomas, delivered a talk titled "Better software for Better research: Introduction to the FAIR2 for Research Software training programme". In this introductory talk he presented what the FAIR principles are and why they are crucial to make your research more open. We gave some initial hints on how to apply them to software and present the FAIR training programme that our teams have designed for the UoS research community.

Reviewer: Very minor...

```suggestion
In the first session which took place in October, our Head of Research Software Engineering, Romain Thomas, delivered a talk titled "_Better software for Better research: Introduction to the FAIR2 for Research Software training programme_". In this introductory talk he presented what the FAIR principles are and why they are crucial to make your research more open. We gave some initial hints on how to apply them to software and present the FAIR training programme that our teams have designed for the UoS research community.
```
